src/main/java/nl/dtls/fairdatapoint/service/openapi/OpenApiService.java

```diff
@@ -0,0 +1,111 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package nl.dtls.fairdatapoint.service.openapi;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import lombok.extern.log4j.Log4j2;
+import nl.dtls.fairdatapoint.database.mongo.repository.ResourceDefinitionRepository;
+import nl.dtls.fairdatapoint.entity.resource.ResourceDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@Log4j2
+public class OpenApiService {
+
+    @Autowired
+    private OpenAPI openAPI;
+
+    @Autowired
+    private OpenApiGenerator openApiGenerator;
+
+    @Autowired
+    private ResourceDefinitionRepository resourceDefinitionRepository;
+
+    private Paths getGenericPaths() {
+        return (Paths) openAPI.getExtensions().get("fdpGenericPaths");
+    }
+
+    private boolean isRelatedToResourceDefinition(PathItem pathItem, ResourceDefinition rd) {
+        String rdUuid = (String) pathItem.getExtensions().getOrDefault("fdpResourceDefinition", "");
+        return rdUuid.equals(rd.getUuid());
+    }
+
+    public void removeGenericPaths(ResourceDefinition rd) {
+        Paths fdpGenericPaths = getGenericPaths();
+        Set<String> toRemove = fdpGenericPaths
+                .entrySet()
+                .stream()
+                .filter(item -> isRelatedToResourceDefinition(item.getValue(), rd))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+        log.info("Removing OpenAPI paths: {}", toRemove);
+        openAPI.getPaths().keySet().removeAll(toRemove);
+        fdpGenericPaths.keySet().removeAll(toRemove);
+    }
+
+    public void removeAllGenericPaths() {
+        Paths fdpGenericPaths = getGenericPaths();
+        log.info("Removing OpenAPI paths: {}", fdpGenericPaths.keySet());
+        openAPI.getPaths().keySet().removeAll(fdpGenericPaths.keySet());
+        fdpGenericPaths.clear();
+        //openAPI.getPaths().keySet().removeAll(openAPI.getPaths().keySet().stream().filter(p -> p.contains("**")).collect(Collectors.toList()));
+    }
+
+    public void updateGenericPaths(ResourceDefinition rd) {
+        Paths fdpGenericPaths = getGenericPaths();
+        // Cleanup
+        removeGenericPaths(rd);
+        // Generate
+        openApiGenerator.generatePathsForResourceDefinition(fdpGenericPaths, rd);
+        // Apply
+        log.info("Adding OpenAPI paths: {}", fdpGenericPaths.keySet());
+        openAPI.getPaths().putAll(fdpGenericPaths);
+    }
+
+    public void updateAllGenericPaths() {
+        Paths fdpGenericPaths = getGenericPaths();
+        // Cleanup
+        removeAllGenericPaths();
```

Comment:
Is this a safeguard, or are the operations marked as `hidden` still in need for removal?

Reply:
I will verify that... `hidden` should be enough.

Reply:
This is actually for clearing the paths added by this service (e.g. if `updateAllGenericPaths` is called multiple times). If there are no generated yet, to does practically nothing.